libbeat/publisher/pipeline/output_test.go

```diff
@@ -227,6 +245,53 @@ func TestMakeClientTracer(t *testing.T) {
 	apmEvents := recorder.Payloads()
 	transactions := apmEvents.Transactions
 	if len(transactions) != numBatches {
+		logger.Flush()
 		t.Errorf("expected %d traces, got %d", numBatches, len(transactions))
 	}
 }
+
+type bufLogger struct {
```

Comment:
why introduce a buffer instead of directly logging to `t`? In order to minimize noise in test output? In that case maybe add a comment explaining the why.

Reply:
Yes, exactly. So we only emit the log entries if the test fails, not if it succeeds. I will add the comment, thanks!

Reply:
Done in 96b39e72d.